data_structures/stacks/evaluate_postfix_notations.py

```diff
@@ -17,32 +17,52 @@ def evaluate_postfix(postfix_notation: list) -> int:
     9
     >>> evaluate_postfix(["4", "13", "5", "/", "+"])
     6
+    >>> evaluate_postfix(["2", "+"])
+    2
+    >>> evaluate_postfix(["5", "-"])
+    -5
     >>> evaluate_postfix([])
     0
     """
     if not postfix_notation:
         return 0
 
     operations = {"+", "-", "*", "/"}
+    unary_operations = {"+"}  # Unary operator(s)
+
     stack: list[Any] = []
 
     for token in postfix_notation:
         if token in operations:
-            b, a = stack.pop(), stack.pop()
-            if token == "+":
-                stack.append(a + b)
-            elif token == "-":
-                stack.append(a - b)
-            elif token == "*":
-                stack.append(a * b)
+            if token == "-":
+                if len(stack) < 1:
+                    raise ValueError(
+                        "Invalid expression: insufficient operands for unary operator"
+                    )
```

Comment:
This doesn't fix the problem with unary negation operators. We want to support negation, not raise an error if it's present.

Reply:
made several changes, look into that.